examples/notedump/notedump_main.c

```diff
@@ -0,0 +1,263 @@
+ * udp_cmdline
+ *****************************************************************************/
+
+bool udp_cmdline(int argc, char **argv)
+{
+  int ret;
+  long port;
+  char *endptr;
+
+  /* Init the default IP address */
+
+  g_udpserver_ipv4 = HTONL(0x01020304);
+  g_udpserver_port = HTONS(1234);
+
+  /* Currently only a single command line option is supported:  The server
+   * IP address. Used to override default.
+   */
+
+  if (argc == 3)
+    {
+      /* Convert the <server-addr> argument into a binary address */
+
+      ret = inet_pton(AF_INET, argv[1], &g_udpserver_ipv4);
+      if (ret == 1)
+        {
+          fprintf(stderr, "Server address: %s\n", argv[1]);
+        }
+      else
+        {
+          fprintf(stderr, "ERROR: Invalid server address: %s\n", argv[1]);
+          show_usage(argv[0]);
+          return false;
+        }
+
+        port = strtol(argv[2], &endptr, 10);
+      if (*endptr != '\0' || port < 1 || port > 65535)
+        {
+          fprintf(stderr, "ERROR: Invalid port: %s\n", argv[2]);
+          show_usage(argv[0]);
+          return false;
+        }
+
+      g_udpserver_port = (uint32_t)port;
+    }
+
+  else if (argc != 1)
+    {
+      fprintf(stderr, "ERROR: Too many arguments\n");
+      show_usage(argv[0]);
+      return false;
+    }
+
+  return true;
+}
+
+/*****************************************************************************
+ * Private Functions
+ *****************************************************************************/
+
+static int create_socket(void)
+{
+  socklen_t addrlen;
+  int sockfd;
+
+  struct sockaddr_in addr;
+
+  /* Create a new IPv4 UDP socket */
+
+  sockfd = socket(AF_INET, SOCK_DGRAM, 0);
+  if (sockfd < 0)
+    {
+      printf("client ERROR: client socket failure %d\n", errno);
+      return -1;
+    }
+
+  /* Bind the UDP socket to a IPv4 port */
+
+  addr.sin_family      = AF_INET;
+  addr.sin_port        = HTONS(CONFIG_EXAMPLES_NOTEDUMP_SERVER_PORTNO);
+  addr.sin_addr.s_addr = HTONL(INADDR_ANY);
+  addrlen              = sizeof(struct sockaddr_in);
+
+  if (bind(sockfd, (FAR struct sockaddr *)&addr, addrlen) < 0)
+    {
+      printf("client ERROR: Bind failure: %d\n", errno);
+      return -1;
+    }
+
+  return sockfd;
+}
+
+/*****************************************************************************
+ * Public Functions
+ *****************************************************************************/
+
+/*****************************************************************************
+ * main
+ *****************************************************************************/
+
+int main(int argc, char **argv, char **envp)
+{
+  /* Parse any command line options */
+
+  if (udp_cmdline(argc, argv) == false)
+    {
+      fprintf(stderr, "ERROR: Invalid arguments\n");
+      return -1;
+    }
+
+  /* Run the server or client, depending upon how target1 was configured */
+
+  struct sockaddr_in server;
+#
```

Comment:
Thanks for contributing the code :-) 
```suggestion

```
Let's remove the `#`. Also let's move the declarations to the top of the `main()` function, like this:
```
int main(int argc, char **argv, char **envp)
{
  struct sockaddr_in server;
  unsigned char outbuf[NOTE_MSG_SIZE];
  ...
```